### PR TITLE
Build static binaries

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

Provide an .deb package of the node.

## Changes

Add 
- dockerfile to build static linux binaries (currently focused on ubuntu)
- debian package template for building `.deb` packages that contain the node and service description.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.

### CODE REVIEW

File: scripts/static-binaries/build-on-ubuntu.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# seem difficult to impossible to link statically. These are libc, libpq.
+
+set -euxo pipefail
+
+# The script must be invoked with the concordium-node source directory as working directory.
+
+# Expected parameters:
+# - EXTRA_FEATURES, features to use in addition to `static` when building the node, e.g., collector. Comma separated list.
+
+extra_features="$EXTRA_FEATURES"
```
Comment (bisgardo) [inline scripts/static-binaries/build-on-ubuntu.sh:14]: As the script doesn't take any args, you could use `$@` to take the args as extra features. As you have `set -u` this setup will crash if `EXTRA_FEATURES` isn't set. You could of course also use `${EXTRA_FEATURES:-""}`.

File: scripts/static-binaries/build-static-binaries.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ubuntu_version="${UBUNTU_VERSION}"
+static_libraries_image_tag="${STATIC_LIBRARIES_IMAGE_TAG}"
+ghc_version="${GHC_VERSION}"
+extra_features=${EXTRA_FEATURES:-""}
+
+docker build \
+	--build-arg ubuntu_version="$ubuntu_version" \
+	--build-arg static_libraries_image_tag="$static_libraries_image_tag" \
+	--build-arg ghc_version="$ghc_version" \
+    --build-arg extra_features="$extra_features" \
```
Comment (bisgardo) [inline scripts/static-binaries/build-static-binaries.sh:14]: indendation (same below)

**Review by bisgardo — APPROVED** [review]
A bit too many things that need manual updates, but looks good otherwise. Great docs!

File: scripts/static-binaries/build-on-ubuntu.sh (hunk 1 above)
Comment (abizjak) [reply scripts/static-binaries/build-on-ubuntu.sh:14]: > As the script doesn't take any args, you could use `$@` to take the args as extra features. As you have `set -u` this setup will crash if `EXTRA_FEATURES` isn't set. You could of course also use `${EXTRA_FEATURES:-""}`.

I'm not sure I agree. The documentation explicitly states that EXTRA_FEATURES is an expected parameter, so crashing if it isn't set should be expected. 

I don't really see adding a default value for it materially helps anything.

File: scripts/static-binaries/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ubuntu. Small modifications of the build scripts will yield binaries for similar
+platforms. The base docker image for the build should be varied and the
+specific package manager for installing dependencies (and possibly renaming dependencies).
+
+## [build-static-binaries.sh](./build-static-binaries.sh)
+    This is the main script that should be used. **It is intended to be run from
+    the root of the repository.** The script supports the following environment variables
+    - `UBUNTU_VERSION`, numeric tag, 20.04, 18.04
+    - `STATIC_LIBRARIES_IMAGE_TAG`, tag of the docker image used to build static
+      libraries. `latest` should work, but otherwise see which tags are
```
[the remark this answers is not in the document]
Comment (abizjak) [reply scripts/static-binaries/README.md:15]: Currently there are versions tagged `0.19` and `latest` on dockerhub.

File: scripts/static-binaries/build-on-ubuntu.sh (hunk 1 above)
Comment (abizjak) [reply scripts/static-binaries/build-on-ubuntu.sh:14]: I see that the documentation in the list above says this is optional, as opposed to the documentation in the script itself. I'll rectify the discrepancy.

File: scripts/distribution/ubuntu-packages/template/debian/postinst [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+if [[ ! -f /etc/systemd/system/concordium-node-collector.service.d/override.conf ]]
+then 
+    echo "Writing node name to '/etc/systemd/system/node-collector.d/override.conf'."
+    mkdir -p /etc/systemd/system/concordium-node-collector.service.d/
+    cat > /etc/systemd/system/concordium-node-collector.service.d/override.conf <<EOF
+[Service]
+Environment="COLLECTOR_NODE_NAME=$NODE_NAME"
+EOF
+else
+    echo "Node collector override file already exists."
```
[the remark this answers is not in the document]
Comment (abizjak) [reply scripts/distribution/ubuntu-packages/template/debian/postinst]: Added.

**Pull request merged**